addons/mail/models/mail_thread.py

```diff
@@ -843,7 +836,21 @@ def _routing_check_route(self, message, message_dict, route, raise_exception=Tru
                 check_result = self.env['mail.alias.mixin']._alias_check_contact_on_record(obj, message, message_dict, alias)
             if check_result is not True:
                 self._routing_warn(_('alias %s: %s') % (alias.alias_name, check_result.get('error_message', _('unknown error'))), message_id, route, False)
-                self._routing_create_bounce_email(email_from, check_result.get('error_template', _generic_bounce_body_html), message)
+
+                lang_author = message_dict.get('author_id') and self.env['res.partner'].browse(message_dict.get('author_id')).lang or False
```

Comment:
Looking a bit at this code, I think we could move it outside of the core gateway. In mail there is a models.py file with some light additions to basemodel. We could add a generic _routing_generate_bounce_body method, overriden in mail.alias mixin, that basically does what is done here. That way we will just do body = obj._routing_generate_bounce_body(). Base behavior in base models is the generic body html, alias mixin tries to use the alias_bounced_content content; and people could override it if wanted.

What do you think ?

Reply:
Well, why not, I am going to push the new version, tell me if I get your point. Also I decided to remove the contact_security from the bounced email to avoid leaking this data to the public, it is good for you ?

Reply:
Yes, people don't need those details :) .

Reply:
There is a limitation if we put the _routing_email(or generate)_bounce_body on the alias mixin: we can't create manually a alias with the feature wanted. It limited the feature scope only for models using the mixin. 
I think that it will be better to put the logic in the alias model ( _get_alias_bounced_body ) instead, it will be not possible to override it in the targeted model side but it is a good trade-off.

Reply:
@tde-banana-odoo

Reply:
It seems you are right. Then what about putting the computation on Basemodel ? That way if there is an alias -> alias yay, no alias -> generic body, and if we want some customization it is possible. Moreover it allows to do obj._get_my_super_bounce_body() and youpla ! What do you think ?